lib/ex_unit/lib/ex_unit/assertions.ex

```diff
@@ -121,6 +121,11 @@ defmodule ExUnit.Assertions do
 
   Even though the match works, `assert` still expects a truth
   value. In such cases, simply use `==/2` or `match?/2`.
+
+  If you need more complex pattern matching using guards, you
+  need to use `match?/2`:
+
+      assert match?([%{id: id} | _] when is_integer(id), records)
```

Comment:
Nit: we usually leave a blank line after code blocks before the closing quotes.

```suggestion
      assert match?([%{id: id} | _] when is_integer(id), records)

```